addons/web_editor/static/lib/odoo-editor/test/spec/html_tables/excel_clipboard.html

```diff
@@ -0,0 +1,200 @@
+                mso-generic-font-family: auto;
+                mso-font-charset: 1;
+                text-align: center;
+            }
+
+            .xl93 {
+                color: red;
+                font-size: 10.0pt;
+                text-decoration: underline;
+                text-underline-style: single;
+                font-family: Arial;
+                mso-generic-font-family: auto;
+                mso-font-charset: 1;
+                text-align: center;
+            }
+
+            .xl94 {
+                color: #495057;
+                font-size: 10.0pt;
+                font-family: "Odoo Unicode Support Noto";
+                mso-generic-font-family: auto;
+                mso-font-charset: 1;
+                text-align: center;
+                background: yellow;
+                mso-pattern: black none;
+            }
+
+            .xl95 {
+                color: red;
+                font-size: 10.0pt;
+                font-family: Arial;
+                mso-generic-font-family: auto;
+                mso-font-charset: 1;
+                text-align: center;
+                background: yellow;
+                mso-pattern: black none;
+                white-space: normal;
+            }
+        </style>
+    </head>
+
+    <body link="#0563C1" vlink="#954F72">
+        <table width=398 style='border-collapse:collapse;width:299pt'><!--StartFragment-->
+            <col width=187 style='width:140pt'>
+            <col width=211 style='width:158pt'>
+            <tr height=20 style='height:15.0pt'>
+                <td width=187 height=20 class=xl88 dir=LTR style='width:140pt;height:15.0pt'><span class=font12>Italic
+                        then also </span><span class=font13>BOLD</span></td>
+                <td width=211 class=xl89 dir=LTR style='width:158pt'><s>Italic strike</s></td>
+            </tr>
+            <tr height=20 style='height:15.0pt'>
+                <td height=20 class=xl90 dir=LTR style='height:15.0pt'><span class=font33>Just bold </span><span
+                        class=font12>Just Italic</span></td>
+                <td class=xl91 dir=LTR>Bold underline</td>
+            </tr>
+            <tr height=20 style='height:15.0pt'>
+                <td height=20 class=xl92 dir=LTR style='height:15.0pt'>Color text</td>
+                <td class=xl93 dir=LTR><s>Color strike and underline</s></td>
+            </tr>
+            <tr height=20 style='height:15.0pt'>
+                <td height=20 class=xl94 dir=LTR style='height:15.0pt'>Color background</td>
+                <td width=211 class=xl95 dir=LTR style='width:158pt'>Color text on color background</td>
+            </tr>
+            <tr height=27 style='height:20.25pt'>
+                <td colspan=2 width=398 height=27 class=xl87 dir=LTR style='width:299pt;height:20.25pt'>14pt MONO TEXT
+                </td>
+            </tr><!--EndFragment-->
+        </table>
+    </body>
+
+    </html>
+</div>
```

Comment:
This file is missing an empty line at the end, GitHub shows this quite visibly.

Reply:
It wasn't easy saving this file _without_ the empty line at the end, but if I have the empty line I'll need to remove it in the code for the tests to pass... What do you think ?

Reply:
Why remove it in the code ? Why not include it in the results ? I mean it would just be an additional text node, right ?

It is likely some people will need to update these files in the future when softwares change the way they generate their html, and I think it's not great that the other people will also struggle to save the file without their IDE creating the empty line automatically.

Reply:
So, even if I add the newline in the clipboard and result files, it doesn't appear in the editor. I'm not sure why that is. I can make the tests pass by adding a newline _also_ in the contentBefore. Just kind of confusing, not sure why the clipboard one is discarded.

Reply:
I'm not sure I understand. What is the particular issue that is specific to excel tests ?

Reply:
I have no idea. Will investigate this afternoon, but feel free to check it out as it's made explicit in my latest changes.

Reply:
So, the answer is nothing is specific to excel tests.
_All_ tests have this behaviour where adding a newline at the _end_ of the "result" files also requires adding a newline at the end of the `beforeContent` (even though the clipboard file also has the newline).
This isn't so weird, the newline from the clipboard is probably discarded.
The really weird thing is that for tests to pass after those modifications I still had to add a newline _before_ the last tag of the "result" file. See my latest force-push.

The reason I didn't notice earlier is that those JS test files are updated in my browser very undeterministically. Like I can go into developper mode, generate test assets, regenerate them, clear my cookies and Ctrl+Shift+R the page and the tests haven't updated (the old version is still being tested), and sometimes they update after a couple more reloads... Only secure way is a _new_ incognito window.

Reply:
I wanted to investigate manually but I can't even do that because my editor just removes the lines with blank spaces and it makes the tests fail anyway. This is precisely what I wanted to avoid, in fact. That other people have a really hard time modifying these files in the future.

Could you investigate where the additional lines come from ? I'm specifically talking about the first two blank lines from the `excel_result` file. I don't understand where they come from. Are they already there when the file is loaded ? I doubt so. Are there already there when we pass it to the past event, somehow the browser mingles with it ? Maybe. Or do we end up creating thoses spaces in `cleanForPaste` or in some other piece of code ? If _we_ are creating them then maybe we could try fixing the code that creates them so that the result files are easier (or possible at all) to edit.

Reply:
So, upon investigation it seems the blank/whitespace lines come from the clipboard files themselves.
- Original clipboard contents have no spacing at all (HTML is in one line)
- The "clipboard" files have been indented automatically with an HTML formatter to ease readability and future modifications
- There is no code that introduces new lines or whitespace, only lack of code that removes them (which seems normal as it's hard to distinguish unimportant indentation from expected content in nodes). For example if an indented tag is removed, its indentation --which is actually part of the parent's text content-- is not.

I pushed a separate commit to simplify the ending of "result" files. However, some still contain blank lines at the beginning of their `<table>` tag, originating from the text content of discarded tags. Those are even dirtier to fix that way (adapting the clipboard content itself).

I would either
- adapt the clipboard content further, perhaps making it messier but avoiding blank/whitespace lines
- put back the original inline clipboard content and result, making it obvious that those files are not meant for manual edition.
- leave it like that and accept that we need editors that support whitespace and blank lines to edit those files
- find a way to remove whitespace/indentation upon reading both the clipboard and result files ?

Awaiting guidance or instructions :)